tests/benchq/resource_estimation/test_extrapolation_estimation.py

```diff
@@ -0,0 +1,268 @@
+import numpy as np
+import pytest
+from orquestra.quantum.circuits import CNOT, RZ, Circuit, H
+
+from benchq.data_structures.hardware_architecture_models import BasicArchitectureModel
+from benchq.data_structures.quantum_program import QuantumProgram
+from benchq.resource_estimation.graph import (
+    ExtrapolationResourceEstimator,
+    GraphResourceEstimator,
+    create_big_graph_from_subcircuits,
+    run_extrapolation_pipeline,
+    run_resource_estimation_pipeline,
+    simplify_rotations,
+    synthesize_clifford_t,
+)
+from benchq.vizualization_tools import plot_extrapolations
+
+
+@pytest.fixture(params=[False, True])
+def use_delayed_gate_synthesis(request):
+    return request.param
+
+
+def _get_transformers(use_delayed_gate_synthesis, error_budget):
+    if not use_delayed_gate_synthesis:
+        transformers = [
+            synthesize_clifford_t(error_budget),
+            create_big_graph_from_subcircuits(
+                delayed_gate_synthesis=use_delayed_gate_synthesis,
+            ),
+        ]
+    else:
+        transformers = [
+            simplify_rotations,
+            create_big_graph_from_subcircuits(
+                delayed_gate_synthesis=use_delayed_gate_synthesis,
+            ),
+        ]
+    return transformers
+
+
+@pytest.mark.parametrize(
+    "quantum_program,steps_to_extrapolate_from,n_measurement_steps_fit_type",
+    [
+        (
+            QuantumProgram(
+                [
+                    Circuit([H(0), RZ(np.pi / 14)(0), CNOT(0, 1)]),
+                    Circuit([H(0), H(1), RZ(np.pi / 14)(0)]),
+                ],
+                5,
+                lambda x: [0] + [1] * x + [0],
+            ),
+            [1, 2, 3, 4],
+            "logarithmic",
+        ),
+        (
+            QuantumProgram(
+                [
+                    Circuit([H(0), RZ(np.pi / 14)(0), CNOT(0, 1)]),
+                    Circuit([H(0), H(1), RZ(np.pi / 14)(0)]),
+                ],
+                1000,
+                lambda x: [0] + [1] * x + [0],
+            ),
+            [1, 2, 3, 5],
+            "logarithmic",
+        ),
+        (
+            QuantumProgram(
+                [
+                    Circuit([H(0), RZ(np.pi / 14)(0), CNOT(0, 1)]),
+                    Circuit([H(0), H(1), RZ(np.pi / 14)(0)]),
+                    Circuit([H(0), H(1), CNOT(0, 1)]),
+                ],
+                100,
+                lambda x: [0] + [1, 2] * x + [0],
+            ),
+            [1, 2, 3, 5, 7, 10, 15, 25],
+            "linear",
+        ),
+    ],
+)
+def test_get_resource_estimations_for_program_gives_correct_results(
+    quantum_program,
+    steps_to_extrapolate_from,
+    use_delayed_gate_synthesis,
+    n_measurement_steps_fit_type,
+):
+    architecture_model = BasicArchitectureModel(
+        physical_gate_error_rate=1e-3,
+        physical_gate_time_in_seconds=1e-6,
+    )
+    error_budget = {
+        "qsp_required_precision": 1e-3,
```

Comment:
I think the two first keys are not really needed anymore.

Reply:
I have rid the world of them.

Reply:
😆